common/lib/xmodule/xmodule/error_module.py

```diff
@@ -30,62 +37,56 @@
 
 class ErrorFields(object):
     """
-    XBlock fields used by the ErrorModules
+    XBlock fields used by the ErrorBlocks
     """
     contents = String(scope=Scope.content)
     error_msg = String(scope=Scope.content)
     display_name = String(scope=Scope.settings)
 
 
-class ErrorModule(ErrorFields, XModule):
+class ErrorBlock(
+    ErrorFields,
+    XModuleDescriptorToXBlockMixin,
+    XModuleToXBlockMixin,
+    HTMLSnippet,
+    ResourceTemplates,
+    XModuleMixin,
+):
     """
     Module that gets shown to staff when there has been an error while
     loading or rendering other modules
     """
 
+    resources_dir = None
+
+    def student_view(self, _context):
+        """
+        Return a fragment that contains the html for the student view.
+        """
+        fragment = Fragment(self.get_html())
+        return fragment
+
     def get_html(self):
```

Comment:
Does `get_html` need to exist at all any longer?

Reply:
Not really. It was being overridden in `NonStaffErrorBlock` below but I have updated it to override `student_view()` instead.